ts/src/pro/binance.ts

```diff
@@ -2143,6 +2153,181 @@ export default class binance extends binanceRest {
         this.handleOrder (client, message);
     }
 
+    async watchPositions (symbols: string[] = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name binance#watchPositions
+         * @description watch all open positions
+         * @param {[string]|undefined} symbols list of unified market symbols
+         * @param {object} params extra parameters specific to the binance api endpoint
+         * @returns {[object]} a list of [position structure]{@link https://docs.ccxt.com/en/latest/manual.html#position-structure}
+         */
+        await this.loadMarkets ();
+        await this.authenticate (params);
+        let market = undefined;
+        let messageHash = '';
+        if (!this.isEmpty (symbols)) {
+            market = this.getMarketFromSymbols (symbols);
+            messageHash = '::' + symbols.join (',');
+        }
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('watchPositions', market, params);
+        if (type === 'swap' || type === 'spot') {
+            type = 'future';
+        }
+        symbols = this.marketSymbols (symbols);
+        messageHash = type + ':positions' + messageHash;
+        const url = this.urls['api']['ws'][type] + '/' + this.options[type]['listenKey'];
+        const client = this.client (url);
+        this.setBalanceCache (client, type);
+        this.setPositionsCache (client, type, symbols);
+        const fetchPositionsSnapshot = this.handleOption ('watchPositions', 'fetchPositionsSnapshot', true);
+        const awaitPositionsSnapshot = this.safeValue ('watchPositions', 'awaitPositionsSnapshot', true);
+        const cache = this.positions[type];
+        if (fetchPositionsSnapshot && awaitPositionsSnapshot && this.isEmpty (cache)) {
+            return await client.future (type + ':fetchPositionsSnapshot');
+        }
+        const newPositions = await this.watch (url, messageHash, undefined, type);
```

Comment:
@pcriadoperez can you check the python version? I changed the position but did not get an update 
![image](https://github.com/ccxt/ccxt/assets/43336371/6e8cec6d-c98e-45ca-90d2-23954a4af70a)

Reply:
@pcriadoperez in Php I see the opposite issue 😅, I get the update but not the snapshot 
![image](https://github.com/ccxt/ccxt/assets/43336371/00a855df-42cf-484e-9fd3-e89b338d7e19)

Reply:
lol.. in php I realized it was because of the is_empty function. I fixed it here (https://github.com/ccxt/ccxt/pull/19173)